src/app.py

```diff
@@ -4212,6 +4213,155 @@ def multiple_components():
     return jsonify(normalized_complete_entity_list)
 
 
+def bulk_update_entities(
+    entity_updates: dict,
+    token: str,
+    entity_api_url: str,
+    total_tries: int = 3,
+    throttle: float = 5,
+    after_each_callback: Optional[Callable[[int], None]] = None,
+) -> dict:
+    """Bulk update the entities in the entity-api.
+
+    This function supports request throttling and retries.
+
+    Parameters
+    ----------
+    entity_updates : dict
+        The dictionary of entity updates. The key is the uuid and the value is the
+        update dictionary.
+    token : str
+        The groups token for the request.
+    entity_api_url : str
+        The url of the entity-api.
+    total_tries : int, optional
+        The number of total requests to be made for each update, by default 3.
+    throttle : float, optional
+        The time to wait between requests and retries, by default 5.
+    after_each_callback : Callable[[int], None], optional
+        A callback function to be called after each update, by default None. The index
+        of the update is passed as a parameter to the callback.
+
+    Returns
+    -------
+    dict
+        The results of the bulk update. The key is the uuid of the entity. If
+        successful, the value is a dictionary with "success" as True and "data" as the
+        entity data. If failed, the value is a dictionary with "success" as False and
+        "data" as the error message.
+    """
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "X-Application": "entity-api",
```

Comment:
We'll need to add a new constant to be used for validation: https://github.com/hubmapconsortium/entity-api/blob/715c7f65b3107ef0859102f7c74a6dc13352755b/src/schema/schema_validators.py#L38

Reply:
Done.